sql/sql_class.h

```diff
@@ -2067,6 +2069,36 @@ class THD : public MDL_context_owner,
   ha_rows m_examined_row_count;
 
  private:
+  std::shared_ptr<explicit_snapshot> m_explicit_snapshot;
+
+ public:
+  /**
+   * Creates an explicit snapshot and associates it with the current conn
+   * return true if error, false otherwise
+   */
+  bool create_explicit_snapshot();
+
+  inline std::shared_ptr<explicit_snapshot> get_explicit_snapshot() {
```

Comment:
inside-class method definitions implicitly means inline, this should be a const method
```suggestion
  explicit_snapshot_ptr get_explicit_snapshot() const {
```